runner/file.go

```diff
@@ -507,6 +509,55 @@ func transformYAMLFile(in string, out *ResultWriter) error {
 	return transformYAML(r, out)
 }
 
+func transformLogFmt(in *bufio.Reader, out *ResultWriter) error {
+	o := map[string]any{}
+	keys := map[uint64]string{}
+
+	var h maphash.Hash
+
+	dec := logfmt.NewDecoder(in)
+	for dec.ScanRecord() {
+		for dec.ScanKeyval() {
+			if dec.Key() != nil {
+				h.Write(dec.Key())
+
+				if key, ok := keys[h.Sum64()]; ok {
+					o[key] = string(dec.Value())
+				} else {
+					key := string(dec.Key())
+					keys[h.Sum64()] = key
+					o[key] = string(dec.Value())
+				}
+
+				h.Reset()
+			}
+		}
+
+		err := out.WriteRow(o)
+		if err != nil {
+			return err
+		}
+
+		o = make(map[string]any)
```

Comment:
This is still an allocation ever loop. You should just omit this.

Reply:
I don't think I should omit it unless inconsistency is acceptable. It will only work assuming that each log has exactly the same set of fields.

> time="2015-03-26T01:27:38-04:00" level=debug msg="Started observing beach" animal=walrus number=8
time="2015-03-26T01:27:38-04:00" level=info msg="A group of walrus emerges from the ocean" animal=walrus size=10
time="2015-03-26T01:27:38-04:00" level=warning msg="The group's number increased tremendously!" number=122 omg=true
time="2015-03-26T01:27:38-04:00" level=debug msg="Temperature changes" temperature=-4
time="2015-03-26T01:27:38-04:00" level=panic msg="It's over 9000!" animal=orca size=9009

In cases like this, keys and values from earlier logs will be copied to consecutive rows without those fields.
In fact it is exactly what happens in the test when you comment the like out.